etc/cmake/options.cmake

```diff
@@ -127,3 +127,8 @@ option(OTBR_DNS_UPSTREAM_QUERY "Allow sending DNS queries to upstream" OFF)
 if (OTBR_DNS_UPSTREAM_QUERY)
     target_compile_definitions(otbr-config INTERFACE OTBR_ENABLE_DNS_UPSTREAM_QUERY=1)
 endif()
+
+option(OTBR_PUBLISH_ID_AS_EXT_ADDRESS "Publish the 'id' TXT entry as the Extended MAC Address, enable this feature only when 'id' is not set via dbus API" OFF)
+if (OTBR_PUBLISH_ID_AS_EXT_ADDRESS)
+    target_compile_definitions(otbr-config INTERFACE OTBR_ENABLE_OTBR_PUBLISH_ID_AS_EXT_ADDRESS=1)
+endif()
```

Comment:
I feel the second `OTBR` in `OTBR_ENABLE_OTBR_PUBLISH_ID_AS_EXT_ADDRESS` is redundant. Also it may worth specifying in the name that this option only applies to `meshcop` service.

Reply:
Thanks @superwhd!

> Also it may worth specifying in the name that this option only applies to meshcop service.

I am updating the help text to say "Publish the MeshCoP mDNS 'id' TXT entry as the Extended MAC Address...", should this help? Or do you have a name suggestion?

Reply:
I'd suggest mentioning `meshcop` in the cmake option and the macro's names. e.g. `OTBR_PUBLISH_MESHCOP_ID_AS_EXT_ADDRESS` and `OTBR_ENABLE_PUBLISH_MESHCOP_ID_AS_EXT_ADDRESS`.

Reply:
But it's `Border Agent/Router ID` rather than `MeshCoP ID`

Reply:
I agree on including `Border Agent` in the names. My point is to clarify the scope that the option applies to. There could be more than one ID concepts among all BR features.

Reply:
Okay, updated to `OTBR_PUBLISH_MESHCOP_ID_AS_EXT_ADDRESS`!